static/app/components/button.tsx

```diff
@@ -31,6 +32,19 @@ interface BaseButtonProps
    * Positions the text within the button.
    */
   align?: 'center' | 'left' | 'right';
+  /**
+   * Used when you want to overwrite the default Reload event key for analytics
+   */
+  analyticsEventKey?: string;
+  /**
+   * Used when you want to send an Amplitude Event. By default, Amplitude events are not sent so
+   * you must pass in a eventName to send an Amplitude event.
+   */
+  analyticsEventName?: string;
```

Comment:
I'm curious how this is expected to be used? Right now we don't pass the event name anywhere, we use the event key which has a mapping to event name in `trackAdvancedAnalyticsEvent.tsx`, which is used like:

```ts
trackAdvancedAnalyticsEvent('search.searched', { organization, query })
```

I don't think we would want to pass in the event name (in this case `Search: Performed Search`). To keep the scope for this PR as small as we can, maybe you can remove these analytics props for this first iteration and add them in later?

Reply:
so by default, these button clicks won't be sent to Amplitude (which used Event Name instead of event key). we auto generate a event key that we send the event to. this gives the option to add your own eventkey or event name. passing something in for analyticsEventKey would override the auto-generated eventKey. passing in something for analyticsEventName would send the event to Amplitude, so it is different from how trackAdvancedAnalyticsEvent is done and is more similar to how automatic page load analytics are being used.